controllers/ovs_controller.go

```diff
@@ -330,6 +334,60 @@ func (r *OVSReconciler) reconcileNormal(ctx context.Context, instance *ovsv1beta
 	}
 	// create DaemonSet - end
 
+	// create OVN Config Job - start
+	if instance.IsReady() {
+		configHash := instance.Status.Hash[ovsv1beta1.OvnConfigHash]
+		jobDef, err := ovs.ConfigJob(ctx, helper, r.Client, instance, inputHash, serviceLabels)
+		if err != nil {
+			r.Log.Error(err, "Failed to create OVN controller configuration Job")
+			return ctrl.Result{}, err
+		}
+		configJob := job.NewJob(
+			jobDef,
+			ovsv1beta1.OvnConfigHash,
+			false,
+			5,
+			configHash,
+		)
+		ctrlResult, err = configJob.DoJob(ctx, helper)
+		if (ctrlResult != ctrl.Result{}) {
+			instance.Status.Conditions.Set(
+				condition.FalseCondition(
+					condition.ServiceConfigReadyCondition,
+					condition.RequestedReason,
+					condition.SeverityInfo,
+					condition.ServiceConfigReadyMessage,
+				),
+			)
+			return ctrlResult, nil
+		}
+		if err != nil {
+			r.Log.Error(err, "Failed to configure OVN controller")
+			instance.Status.Conditions.Set(
+				condition.FalseCondition(
+					condition.ServiceConfigReadyCondition,
+					condition.RequestedReason,
+					condition.SeverityInfo,
+					condition.ServiceConfigReadyErrorMessage,
+					err.Error(),
+				),
+			)
+			return ctrl.Result{}, err
+		}
+		if configJob.HasChanged() {
+			instance.Status.Hash[ovsv1beta1.OvnConfigHash] = configJob.GetHash()
+			if err := r.Client.Status().Update(ctx, instance); err != nil {
+				return ctrl.Result{}, err
+			}
+			r.Log.Info(fmt.Sprintf("Job %s hash added - %s", jobDef.Name, instance.Status.Hash[ovsv1beta1.OvnConfigHash]))
+		}
+		instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
```

Comment:
this needs a  recent lib-common rebase to use helper.Patch() iinstead of calling Update(). Also, once the instance updated, we need to exit the reconcile loop to avoid these below changes getting lost (the stale reads problem)

Reply:
I changed to use helper.PatchInstance(). Regarding exiting I'm not sure if I understand correctly Your comment. I moved this helper.PatchInstance() call to defer function so it will be done at the end of the reconcile loop. Is that what You were talking about or did I misunderstood something?